### PR TITLE
Documentation for method of (non-)linear systems

### PR DESCRIPTION
**Pull request by AnHeuermann**
Added some documentation in generated C code and in the runtime for
`linearSystemData[index].method = 0` or `=1`.

### CODE REVIEW

**Comment by AnHeuermann** [comment]
@kabdelhak I think my documentation is correct, because if there is no Jacobian at code generation time available then Susan will generate `method=0` and `method=1` else.

But for models like `ScalableTestSuite.Electrical.DistributionSystemAC.ScaledExperiments.DistributionSystemLinear_N_20_M_20` build with `--generatedSymbolicJacobian` the generated C code has `linearSystemData[0].method=0`, but there is a symbolic Jacobian and it will be used.
It's still unclear to me...

**Comment by kabdelhak** [comment]
> @kabdelhak I think my documentation is correct, because if there is no Jacobian at code generation time available then Susan will generate `method=0` and `method=1` else.
> 
> But for models like `ScalableTestSuite.Electrical.DistributionSystemAC.ScaledExperiments.DistributionSystemLinear_N_20_M_20` build with `--generatedSymbolicJacobian` the generated C code has `linearSystemData[0].method=0`, but there is a symbolic Jacobian and it will be used.
> It's still unclear to me...

The function that does this seems to be `functionInitialLinearSystemsTemp`, from what i see you are right, maybe the value gets changed somewhere else. Try adding a comment after the line in question and see which of those cases it actually is.

File: OMCompiler/Compiler/Template/CodegenC.tpl [hunk 1]
```diff
@@ -1970,7 +1970,7 @@ template functionInitialLinearSystemsTemp(list<SimEqSystem> linearSystems, Strin
            linearSystemData[<%ls.indexLinearSystem%>].equationIndex = <%ls.index%>;
            linearSystemData[<%ls.indexLinearSystem%>].size = <%size%>;
            linearSystemData[<%ls.indexLinearSystem%>].nnz = <%nnz%>;
-           linearSystemData[<%ls.indexLinearSystem%>].method = 1;
+           linearSystemData[<%ls.indexLinearSystem%>].method = 1;   /* Symbolic Jacobian avaialble */
```
Comment (lochel) [inline OMCompiler/Compiler/Template/CodegenC.tpl]: typo: available

File: OMCompiler/Compiler/Template/CodegenC.tpl [hunk 2]
```diff
@@ -1953,7 +1953,7 @@ template functionInitialLinearSystemsTemp(list<SimEqSystem> linearSystems, Strin
            linearSystemData[<%ls.indexLinearSystem%>].equationIndex = <%ls.index%>;
            linearSystemData[<%ls.indexLinearSystem%>].size = <%size%>;
            linearSystemData[<%ls.indexLinearSystem%>].nnz = <%nnz%>;
-           linearSystemData[<%ls.indexLinearSystem%>].method = 0;
+           linearSystemData[<%ls.indexLinearSystem%>].method = 0;   /* No symbolic Jacobian avaialble */
```
Comment (lochel) [inline OMCompiler/Compiler/Template/CodegenC.tpl]: typo: available

File: OMCompiler/Compiler/Template/CodegenC.tpl (hunk 1 above)
Comment (AnHeuermann) [reply OMCompiler/Compiler/Template/CodegenC.tpl]: Thanks for finding. I'll update it.

Comment (lochel) [reply OMCompiler/Compiler/Template/CodegenC.tpl]: I really appreciate these kind of comments 👍

**Pull request merged**